agent/containers/images/visualizers/run.py

```diff
@@ -28,9 +28,15 @@
 # Start the collector (if needed)
 collector_cmd = os.environ["COLLECTOR"]
 if not collector_cmd == "":
-    collector = subprocess.Popen(collector_cmd.split(" "))
+    cmd = collector_cmd.split(" ")
     if os.environ["VIS_TYPE"] == "pcp":
-        subprocess.Popen("redis-server")
+        if not os.environ["REDIS_HOST"] == "":
+            cmd[4] = f"--redishost={os.environ['REDIS_HOST']}"
```

Comment:
This assumption about the order of the options seems fragile. It's almost like we really want a dict of option to value pairs so that you can rebuild the value of the `--redishost` option without making this sort of assumption.

For example, instead of a single `COLLECTOR` env, you could have separate variables for each option, which you could replace easily as you build a command from the pieces. Or it wouldn't be all that hard to parse the `COLLECTOR` into a dict (you could even use `ArgumentParser` to do it).

But short of that effort, maybe just a clear comment in the dockerfile that the _order_ of options in the `COLLECTOR` command env is critical and must be retained?